tests/core/test_solve.py

```diff
@@ -461,11 +452,17 @@ def test_only_deps_2():
             'channel-1::meta-0.4.2.dev-py27_0',
             'channel-1::nose-1.3.0-py27_0',
             'channel-1::numpy-1.7.1-py27_0',
-            # 'channel-1::numba-0.5.0-np17py27_0',
+            # 'channel-1::numba-0.5.0-np17py27_0',  # not in the order because only_deps
         )
         assert convert_to_dist_str(final_state_2) == order
 
+    # fails because numpy=1.5 is in our history as an explicit spec
     specs_to_add = MatchSpec("numba=0.5"),
+    with get_solver(specs_to_add, prefix_records=final_state_1, history_specs=specs) as solver:
```

Comment:
These failures are new behavior in this PR.  We're much more strict about requiring people to explicitly override any past explicit decision.